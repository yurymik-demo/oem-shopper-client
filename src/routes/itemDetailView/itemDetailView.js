import React from "react";
import { connect } from "react-redux";
import ItemDetailCard from "../../components/ItemDetailCard/ItemDetailCard";
import { getItemDetails } from "../../actions/items";

export class ItemDetailView extends React.Component {
  componentDidMount() {
    this.props.getItemDetails(this.props.itemId);
  }

  render() {
    const { item } = this.props;
    return (
      <div>
        <h1>Item Details: </h1>
        <ItemDetailCard item={item} />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  item: state.itemDetail,
  itemId: props.match.params.id
});

export default connect(
  mapStateToProps,
  { getItemDetails }
)(ItemDetailView);
