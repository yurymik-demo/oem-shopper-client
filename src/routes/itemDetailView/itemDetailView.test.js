import React from "react";
import renderer from "react-test-renderer";
import { ItemDetailView } from "./itemDetailView";

jest.mock("../../components/ItemsList/ItemsList", () => "ItemsListComponent");
jest.mock(
  "../../components/ItemDetailCard/ItemDetailCard",
  () => "ItemDetailCard"
);

it("renders without crashing", () => {
  const getItemDetailsMock = jest.fn(() => {});
  const wrapper = renderer
    .create(<ItemDetailView getItemDetails={getItemDetailsMock} />)
    .toJSON();
  expect(wrapper).toMatchSnapshot();
  expect(getItemDetailsMock).toHaveBeenCalled();
});
