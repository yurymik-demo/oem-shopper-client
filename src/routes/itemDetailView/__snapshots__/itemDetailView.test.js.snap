// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h1>
    Item Details: 
  </h1>
  <ItemDetailCard />
</div>
`;
